### PR TITLE
Added 'onlyFailed' prop to ScorecardsCard

### PR DESCRIPTION
**Pull request by grantila**
## Improve the usability of ScorecardsCard

This PR adds a prop to the `ScorecardsCard` component, called `onlyFailed`. Before this PR, you can only show either all checks (which can be many, causing a very large/tall card), or specify an exact set of checks to display. With this PR, you can choose to show all kinds of checks, but only those that fail. This is sometimes more valuable. When using `onlyFailed`, if no checks fail, instead of a warning message, an information message is displayed.

For this, the `ScorecardInfo` component now has a prop `noWarning` which turns warnings into informations. This will be default when using `onlyFailed` in the `ScorecardsCard`:

**Before**
Skärmavbild 2024-03-14 kl  14 50 03

**After**
Skärmavbild 2024-03-15 kl  09 23 14

Additionally, the `CheckResultRenderer` type has a property `isFailed` that can be used to identify failing checks. Tech Insights by default only provides a `BooleanCheck`, but implementors can create their own. With this property, they can provide means to identify failed vs passed checks, e.g. for gauges, numbers etc. This PR includes an `isFailed` implementation for the built-in `BooleanCheck`.

### Usage

You provide this prop when using `ScorecardsCard`, most likely in the `EntityPage.tsx`:

```tsx
<EntityTechInsightsScorecardCard title="Scorecard" onlyFailed />
```

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by benjdlambert** [comment]
@Xantier can I get a quick +1 here too? :pray:

File: .changeset/twelve-doors-speak.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-tech-insights': minor
```
Comment (Xantier) [inline .changeset/twelve-doors-speak.md:2]: Should this be patch instead? We are still below 1.x

**Review by Xantier — COMMENTED** [review]
Thanks! Looks grand and can be merged after changeset downgrade to patch

**Review by Xantier — APPROVED** [review]
LGTM

**Review by benjdlambert — APPROVED** [review]
Thanks @grantila, great to see you around :eyes: :pray:

**Pull request merged**